crypto/fipsmodule/rsa/rsa.c

```diff
@@ -902,7 +902,7 @@ DEFINE_LOCAL_DATA(BIGNUM, g_small_factors) {
   out->flags = BN_FLG_STATIC_DATA;
 }
 
-int RSA_check_fips(RSA *key) {
+int RSA_check_fips(const RSA *key) {
```

Comment:
But `key` is an argument to `RSA_sign()` which would then drop const'ness? https://github.com/awslabs/aws-lc/blob/main/crypto/fipsmodule/rsa/rsa.c#L969

Reply:
The documentation for `RSA_check_fips` claims:
> This function does not mutate |rsa| for thread-safety purposes and may be used concurrently.

Should the parameter to `RSA_sign` also be `const RSA*`?  Or does `RSA_sign` modify the parameter?

Reply:
I'm just speaking from a function signature compatibility perspective. The signature doesn't match.

These functions doesn't mutate the `key` and you should be able to change them. But that would likely make a lot of applications angry.

Reply:
Ok. So for historical reasons, the function signature allows for mutation of the parameter even though the implementation currently doesn't.